opal/clearwater/component.rb

```diff
@@ -136,15 +259,20 @@ def self.sanitize_attributes attributes
         attributes[:class_name] = attributes.delete(:class)
       end
 
-      attributes.each do |key, handler|
-        if key[0, 2] == 'on'
-          attributes[key] = proc do |event|
-            handler.call(Bowser::Event.new(event))
+      memo = {attributes: attributes[:attributes] || {} }
+      sanitized = attributes.each_with_object(memo) do |(key, value), hash|
```

Comment:
As much as I would love not to modify the original hash, it creates a _lot_ of GC pressure if we create a new one. Memory usage in the [Clearwater DBMon app](http://clearwater-dbmonster.herokuapp.com) went from 30MB up to 50MB when I tried to make almost this exact same change. An increase in memory usage of 67% is more than I think is reasonable here.

I think the cases where someone passes in a hash that they want to keep unmodified are rare. I can't think of a reason anyone might want to reuse this hash elsewhere, so it should be safe to keep this as-is. We just need to communicate that the hash passed in no longer belongs to them. :-)